examples/optional_redirect/src/tests.rs

```diff
@@ -0,0 +1,38 @@
+use super::rocket;
+use rocket::testing::MockRequest;
+use rocket::http::{Method, Status};
+
+fn test_200(uri: &str, expected_status: Status, expected_body: String) {
+    let rocket = rocket::ignite()
+        .mount("/", routes![super::root, super::user, super::login]);
+    let mut request = MockRequest::new(Method::Get, uri);
+    let mut response = request.dispatch_with(&rocket);
+
+    assert_eq!(response.status(), expected_status);
+    assert_eq!(response.body().and_then(|b| b.into_string()), Some(expected_body));
+}
+
+fn test_303(uri: &str, expected_status: Status, expected_location: &str) {
```

Comment:
This function is called `test_303`, so you shouldn't need the `expected_status` parameter. Test for Status::SeeOther directly below instead.